src/main/java/kostka/moviecatalog/service/MovieDetailService.java

```diff
@@ -0,0 +1,65 @@
+package kostka.moviecatalog.service;
+
+import kostka.moviecatalog.entity.Comment;
+import kostka.moviecatalog.entity.Movie;
+import kostka.moviecatalog.entity.MovieComments;
+import kostka.moviecatalog.entity.MovieDetail;
+import kostka.moviecatalog.exception.MovieNotFoundException;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+import org.springframework.web.client.RestTemplate;
+
+import java.util.List;
+import java.util.Objects;
+
+@Service
+public class MovieDetailService {
+    private DbMovieService movieService;
+    private static final Logger LOGGER = LoggerFactory.getLogger(MovieDetailService.class);
+    private static final String COMMENT_URL = "http://localhost:8082/comments/";
+    private RestTemplate restTemplate = new RestTemplate();
+
+    @Autowired
+    public MovieDetailService(final DbMovieService movieService) {
+        this.movieService = movieService;
+    }
+
+    public MovieDetail getMovieDetail(final Long movieId) {
+        Movie movie = null;
+        try {
+            movie = movieService.getMovie(movieId);
+        } catch (MovieNotFoundException e) {
+            LOGGER.error("Movie not found", e);
+            movie = new Movie();
+            movie.setName("Movie does not exists  with this id: " + movieId);
+        }
+        List<Comment> comments = this.getCommentsFromCommentService(movieId);
+
+        if (movie == null || comments == null) {
```

Comment:
Kdyz je comments null, tak se vyhodi exception? To mi moc nedava smysl. Neslo by udelat, aby `getCommentsFromCommentService ` vracelo prazdny seznam?
Jinak jak muze byt moview null, kdyz se v catch bloku na radku 35 vytvari prazdne movie?

Reply:
comments nebude nikdy null - zapomnel jsem na tuto podminku po refactoringu .. 
Díky! :)